src/models/event-timeline-set.ts

```diff
@@ -180,6 +217,15 @@ export class EventTimelineSet extends TypedEventEmitter<EmittedEvents, EventTime
         return this.liveTimeline;
     }
 
+    /**
+     * Set the live timeline for this room.
+     *
+     * @return {module:models/event-timeline~EventTimeline} live timeline
+     */
+    public setLiveTimeline(timeline: EventTimeline): void {
+        this.liveTimeline = timeline;
+    }
```

Comment:
How are downstream consumers which may be holding a reference to live timeline meant to know it got updated from the context of an EventTimelineSet (feels breaking)

Reply:
Clients which choose to implement timeline refreshing can also listen for `RoomEvent.TimelineRefresh`.

For example, Element listens for `RoomEvent.TimelineRefresh` and tells `TimelineWindow` to load again which picks up from the same `timelineSet`